agent/grpc-external/services/resource/read.go

```diff
@@ -0,0 +1,54 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/consul/internal/resource"
+	"github.com/hashicorp/consul/internal/storage"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+func (s *Server) Read(ctx context.Context, req *pbresource.ReadRequest) (*pbresource.ReadResponse, error) {
+
+	// check type exists
+	_, ok := s.registry.Resolve(req.Id.Type)
+	if !ok {
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("resource type %s not registered", resource.ToGVK(req.Id.Type)))
+	}
+
+	readFn := s.backend.Read
+	if isConsistentRead(ctx) {
+		readFn = s.backend.ReadConsistent
+	}
+
+	resource, err := readFn(ctx, req.Id)
```

Comment:
So sorry, but I've just broken this API in #16538 😅 

It should now be:
```suggestion
	consistency := storage.EventualConsistency
	if isConsistentRead(ctx) {
		consistency = storage.StrongConsistency
	}

	resource, err := s.backend.Read(ctx, consistency, req.Id)
```

Reply:
Fixed